qsr_lib/src/qsrlib_qsrs/qsr_rcc3_rectangle_bounding_boxes_2d.py

```diff
@@ -90,13 +90,83 @@ def __return_all_possible_combinations(self, objects_names):
         return ret
 
     def __compute_qsr(self, bb1, bb2):
-        count_occluded_points = 0
-        for i in range(0, len(bb2), 2):
-            if self.__is_point_in_rectangle([bb2[i], bb2[i+1]], bb1):
-                count_occluded_points += 1
-        results = {0: "dc", 1: "po", 2: "o"} # pythonic case
-        ret = results[count_occluded_points]
-        return ret
+        """Return symmetrical RCC3 relation
+
+        :param bb1: diagonal points coordinates of first bounding box (x1, y1, x2, y2)
+        :param bb2: diagonal points coordinates of second bounding box (x1, y1, x2, y2)
+        :return: an RCC3 relation from the following: 'dc':disconnected, 'po':partial overlap, 'o': occluded/part of
+        """
+
+        bboxes_intercept_v, rabx, raxPrbx, raby, rayPrby = self.__bboxes_intercept(bb1, bb2)
+        if bboxes_intercept_v:
+            if rabx > 0.0 or raby > 0:
+                return "po"
+            else:
+                occluded_points = self.__count_occluded_points(bb1, bb2)
+                if occluded_points >= 4:
+                    return "o"
+                else:
+                    return "po"
+        else:
+            return "dc"
+
+
+    def __count_occluded_points(self, bb1, bb2):
+        occluded_points = 0
+        bb1_4corners = ((bb1[0], bb1[1]),
+                        (bb1[2], bb1[1]),
+                        (bb1[2], bb1[3]),
+                        (bb1[0], bb1[3]))
+        bb2_4corners = ((bb2[0], bb2[1]),
+                        (bb2[2], bb2[1]),
+                        (bb2[2], bb2[3]),
+                        (bb2[0], bb2[3]))
+
+        for p in bb1_4corners:
+            if self.is_point_in_rectangle(p, bb2):
```

Comment:
This is missing the two leading underscores. This is only called if you have overlapping objects. By adding the two underscores it seems to work